platform/m6/sub/101a/boot.c

```diff
@@ -0,0 +1,321 @@
+			"    bl      park_current_core_then_continue\n"
+			"loc_boot:\n"
+			"    mrc     p15, #0, r0, c0, c0, #5\n"
+			"    ands    r0, #0xf\n"
+			"    bl      setup_mmu_for_chdk\n"
+	);
+
+	// capdis -f=chdk -stubs -s=0xe0020025 -c=59 PRIMARY.BIN 0xe0000000
+	asm volatile (
+			"    mrc     p15, #0, r5, c0, c0, #5\n"
+			"    ands    r0, r5, #0xf\n"
+			"    bne     loc_e0020032\n"
+			"    b.w     loc_e002003c\n"
+			"loc_e0020032:\n"
+			"    b.w     sub_e00200f8_my\n"
+			"loc_e002003c:\n"
+			"    ldr     r0, =0xe115d544\n"
+			"    ldr     r1, =0x00008000\n"
+			"    ldr     r3, =0x00068524\n"
+			"loc_e0020042:\n"
+			"    cmp     r1, r3\n"
+			"    itt     lo\n"
+			"    ldrlo   r2, [r0], #4\n"
+			"    strlo   r2, [r1], #4\n"
+			"    blo     loc_e0020042\n"
+			"    ldr     r1, =0x002b4138\n"
+			"    mov.w   r2, #0\n"
+			"loc_e0020056:\n"
+			"    cmp     r3, r1\n"
+			"    it      lo\n"
+			"    strlo   r2, [r3], #4\n"
+			"    blo     loc_e0020056\n"
+			"    ldr     r0, =0xe11bda68\n"
+			"    ldr     r1, =0x01900000\n"
+			"    ldr     r3, =0x01901444\n"
+			"loc_e0020066:\n"
+			"    cmp     r1, r3\n"
+			"    itt     lo\n"
+			"    ldrlo   r2, [r0], #4\n"
+			"    strlo   r2, [r1], #4\n"
+			"    blo     loc_e0020066\n"
+			"    ldr     r0, =0x01900000\n"
+			"    ldr     r1, =0x00001444\n"
+			"    bl      _dcache_clean_by_mva\n"
+			"    ldr     r0, =0x01900000\n"
+			"    ldr     r1, =0x00001444\n"
+			"    bl      _icache_branchpr_invalidate\n"
+			"    ldr     r0, =0xe11beeac\n"
+			"    ldr     r1, =0xdffc4900\n"
+			"    ldr     r3, =0xdffd9ba0\n"
+			"loc_e002008a:\n"
+			"    cmp     r1, r3\n"
+			"    itt     lo\n"
+			"    ldrlo   r2, [r0], #4\n"
+			"    strlo   r2, [r1], #4\n"
+			"    blo     loc_e002008a\n"
+			"    ldr     r1, =0xdffd9ba0\n"
+			"    mov.w   r2, #0\n"
+			"loc_e002009e:\n"
+			"    cmp     r3, r1\n"
+			"    it      lo\n"
+			"    strlo   r2, [r3], #4\n"
+			"    blo     loc_e002009e\n"
+
+			// install CreateTask patch
+			// use half words in case source or destination not word aligned
+			"    adr     r0, patch_CreateTask\n"// src: patch data
+			"    ldr     r1, =hook_CreateTask\n"// dest: address to patch
+			"    add     r2, r0, #10\n"// 2.5 words as target is not word aligned
+			"patch_hook_loop:\n"
+			"    ldrh   r3, [r0],#2\n"
+			"    strh   r3, [r1],#2\n"
+			"    cmp    r0,r2\n"
+			"    blo    patch_hook_loop\n"
+
+			"    ldr     r0, =0xdffc4900\n"
+			"    ldr     r1, =0x000152a0\n"
+			"    bl      _dcache_clean_by_mva\n"
+			"    ldr     r0, =0xdffc4900\n"
+			"    ldr     r1, =0x000152a0\n"
+			"    bl      _icache_branchpr_invalidate\n"
+			"    b       loc_e0020032\n"// +
+
+			"    .align  2\n"
+			"    .short 0\n"// added for alignment
+			"patch_CreateTask:\n"
+			"    ldr.w   pc, _createtask_my\n"// jump to absolute address CreateTask_my
+			"    .short 0\n"// added for alignment
+			"_createtask_my:\n"
+			"    .long   CreateTask_my + 1\n"// has to be a thumb address
```

Comment:
@srsa-hu done the file modules; CHDK starts straightaway!

I'm staring at the M100's boot.c and I can't find where you are unlocking the card and enabling the ALT button...

Reply:
You'll need to implement and override the keyboard task (mykbd_task) for them to work. Note that 1) you don't need the CAM_HAS_JOGDIAL parts for basic operation 2) you won't have control over the shutter button(s) until you implement the PhySw_ReleaseSw stuff. The workaround I use in kbd.c for the shoot/halfshoot states is not very good.